db/sql/sql.go

```diff
@@ -1299,3 +1315,48 @@ func (db *DB) SetMaxOpenConns(maxIdleConns int) {
 func safeMysqlContext(_ context.Context) context.Context {
 	return context.Background()
 }
+
+type Like map[string]interface{}
+
+func (lk Like) ToSql() (sql string, args []interface{}, err error) {
+	var (
+		exprs []string
+		opr   = "LIKE"
+	)
+
+	for key, val := range lk {
+		expr := ""
+
+		switch v := val.(type) {
+		case driver.Valuer:
+			if val, err = v.Value(); err != nil {
+				return
+			}
+		}
+
+		if val == nil {
+			err = fmt.Errorf("cannot use null with like operators")
+			return
+		} else {
```

Comment:
This `else {` is not needed, we always return from the function when taking the first branch.

Reply:
else where the ESCAPE injunction is

Reply:
I'm not asking to remove the entire `else` block. Let me explain using an example.

We have a similar situation to the following:
```go
if cond1 {
  return
} else {
  if cond2 {
    return
  } else {
    doStuff()
  }
}
```

This is equivalent to:
```go
if cond1 {
  return
}

if cond2 {
  return
}

doStuff()
```

Doing this transformation reduces unneeded indentation, which with deep nesting, makes it hard to follow the code.